src/dispatch/notification/service.py

```diff
@@ -66,3 +74,31 @@ def delete(*, db_session, notification_id: int):
     )
     db_session.delete(notification)
     db_session.commit()
+
+
+def send(*, db_session, class_instance, notification_params):
+    """Sends notifications."""
+    notifications = get_all(db_session=db_session)
+    for notification in notifications:
+        for filter in notification.filters:
+            match = search_service.match(
+                db_session=db_session,
+                filter_spec=filter,
+                class_instance=class_instance,
+            )
+            if match and notification.enabled:
+                plugin = plugin_service.get_active(
+                    db_session=db_session, plugin_type=notification.type
```

Comment:
@kevgliss I'm using `notification.type` here to avoid having a bunch of ifs (e.g. for conversation, email, etc). We will need to make sure that when users create or edit notifications in the UI, the type matches the plugin types we support.

Reply:
We should probably create a plugin pydantic model that will enforce this for us since we have the same requirement for service (only allowing plugins we support).